rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +228,49 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+        SyncUtils syncUtils = new SyncUtils(RDTApplication.getInstance());
+        UserAuthorizationVerificationTask userAuthorizationVerificationTask = UserAuthorizationVerificationTask.getInstance(syncUtils);
+        if (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.PENDING)) {
+            userAuthorizationVerificationTask.execute();
+        } else if (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.FINISHED)) {
+            userAuthorizationVerificationTask.destroyInstance();
+            UserAuthorizationVerificationTask.getInstance(syncUtils).execute();
+        }
+    }
+
+    public static class UserAuthorizationVerificationTask extends AsyncTask<Void, Void, Void> {
+
+        private static UserAuthorizationVerificationTask INSTANCE;
+        private final SyncUtils syncUtils;
+
+        public static UserAuthorizationVerificationTask getInstance(SyncUtils syncUtils) {
```

Comment:
Remove the syncutils from this and have `Context` as an argument

Reply:
Instantiate syncutils within the constructor.

Reply:
fixed.